storage/innobase/srv/srv0srv.cc

```diff
@@ -3542,6 +3542,16 @@ srv_enable_redo_encryption()
 bool
 srv_enable_redo_encryption_mk()
 {
+	switch (existing_redo_encryption_mode) {
+	case REDO_LOG_ENCRYPT_RK:
+		ib::error() << "Redo log encryption mode can't be switched without "
```

Comment:
Add error code IB_XB_ and also change the message to print the current and desired redo modes. "Cannot change from X to Y"

IMHO, the warning sent to client should be converted to error.

Reply:
I don't think 5.7 ib::error supports this - all existing uses are with hardcoded strings, error codes were only added in 8.0.